crates/net/discv4/src/lib.rs

```diff
@@ -578,41 +647,128 @@ impl Discv4Service {
         let key = kad_key(node_id);
         let removed = self.kbuckets.remove(&key);
         if removed {
+            debug!(target: "discv4", ?node_id, "removed node");
             self.notify(DiscoveryUpdate::Removed(node_id));
         }
         removed
     }
 
-    /// Updates the node entry
-    fn update_node(&mut self, record: NodeRecord) {
-        if record.id == self.local_enr.id {
+    /// Update the entry on RE-ping
+    ///
+    /// On re-ping we check for a changed enr_seq if eip868 is enabled and when it changed we sent a
+    /// followup request to retrieve the updated ENR
+    fn update_on_reping(&mut self, record: NodeRecord, last_enr_seq: Option<u64>) {
+        if record.id == self.local_node_record.id {
             return
         }
+
+        let last_enr_seq = if let Some(last_enr_seq) = last_enr_seq {
+            last_enr_seq
+        } else {
+            // no need to check for increased enr seq
+            let _ = self.insert_or_update(record, None);
+            return
+        };
+
+        let key = kad_key(record.id);
+
+        let mut fork_id = None;
+        let mut unchanged = false;
+        if let kbucket::Entry::Present(entry, _) = self.kbuckets.entry(&key) {
+            let value = entry.value();
+            // we want to keep the forkid if it was set previously and only update the timestamp
+            if let Some(current_enr) = value.last_enr_seq {
+                if last_enr_seq == current_enr {
+                    unchanged = true;
+                    // ENR sequence is unchanged, so we can keep the fork id
+                    fork_id = value.fork_id;
+                }
+            }
+        }
+
+        // update the value but keep the existing forkid
+        if unchanged || fork_id.is_some() {
+            let entry = NodeEntry {
+                record,
+                last_seen: Instant::now(),
```

Comment:
We need to update the timestamp to prevent additional pinging because we now know the node still exists. If we didn't do this we'd probably end up sending redundant pings o rsooner than we need to.